core/java/iesi-rest-without-microservices/src/test/java/io/metadew/iesi/server/rest/configuration/security/jwt/JWTAuthenticationFilterTest.java

```diff
@@ -0,0 +1,82 @@
+        assertThat(response.getStatus()).isEqualTo(401);
+        assertThat(response.getContentAsString()).contains("com.auth0.jwt.exceptions.JWTDecodeException: The string");
+    }
+
+    @Test
+    public void testInvalidClaims() throws Exception {
+
+        String token = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE0Nzc1OTJ9.isvT0Pqx0yjnZk53mUFSeYFJLDs-Ls9IsNAm86gIdZo";
+        MockHttpServletRequest request = new MockHttpServletRequest("GET", "/scripts");
+        request.addHeader(HttpHeaders.AUTHORIZATION, "Bearer " + token);
+        JWTAuthenticationFilter jwtAuthenticationFilter = new JWTAuthenticationFilter(jwtAuthenticationConverter);
+        MockHttpServletResponse response = new MockHttpServletResponse();
+        MockFilterChain chain = new MockFilterChain();
+
+        jwtAuthenticationFilter.doFilterInternal(request, response, chain);
+        assertThat(response.getStatus()).isEqualTo(401);
+        assertThat(response.getContentAsString()).contains("com.auth0.jwt.exceptions.InvalidClaimException: The Claim");
+    }
+
+    @Test
+    public void testSignatureVerificationException() throws Exception {
+
+        String token = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXUyJ9.eyJpc3MiOiJhdXRoMCJ9.AbIJTDMFc75MhvcP03nJPyCPzZtQcGEp-zWfOkEE";
+        MockHttpServletRequest request = new MockHttpServletRequest("GET", "/scripts");
+        request.addHeader(HttpHeaders.AUTHORIZATION, "Bearer " + token);
+        JWTAuthenticationFilter jwtAuthenticationFilter = new JWTAuthenticationFilter(jwtAuthenticationConverter);
+        MockHttpServletResponse response = new MockHttpServletResponse();
+        MockFilterChain chain = new MockFilterChain();
+
+        jwtAuthenticationFilter.doFilterInternal(request, response, chain);
+        assertThat(response.getStatus()).isEqualTo(401);
+        assertThat(response.getContentAsString()).contains("com.auth0.jwt.exceptions.SignatureVerificationException: The Token's Signature resulted invalid when verified using the Algorithm: HmacSHA256");
+    }
+
+    @Test
+    public void testExpiredToken() throws Exception {
+        String token = "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJpZXNpIiwiZXhwIjoxNTk4NjA2NTY4fQ.F7GnAfcklY1-0sG9K4tkz3mfVQ40aAtBG7MUdSI79FI";
+        MockHttpServletRequest request = new MockHttpServletRequest("GET", "/scripts");
+        request.addHeader(HttpHeaders.AUTHORIZATION, "Bearer " + token);
+        JWTAuthenticationFilter jwtAuthenticationFilter = new JWTAuthenticationFilter(jwtAuthenticationConverter);
+        MockHttpServletResponse response = new MockHttpServletResponse();
+        MockFilterChain chain = new MockFilterChain();
+
+        jwtAuthenticationFilter.doFilterInternal(request, response, chain);
+        assertThat(response.getStatus()).isEqualTo(401);
+        assertThat(response.getContentAsString()).contains("com.auth0.jwt.exceptions.TokenExpiredException: The Token has expired ");
```

Comment:
is it possible to do a jsonPath assertion on the json string? As can you check that the 'message' field starts with "com.auth0..." and the 'status' field is equal to 401.